types/meteor/test/globals/meteor-tests.ts

```diff
@@ -230,25 +230,39 @@ Posts.insert({ title: "Hello world", body: "First post" });
  assert(Scratchpad.find({number: {$lt: 9}}).count() === 5);
  **/
 
+type DBAnimal = {
+    _id?: string;
+    name: string;
+    sound: string;
+}
+interface Animal extends DBAnimal {}
+
 class Animal {
     constructor(public doc: any) {}
+    makeNoise() {
+        return 'roar';
+    }
 }
 
 interface AnimalDAO {
     _id?: string;
     name: string;
     sound: string;
-    makeNoise?: () => void;
+    makeNoise: () => void;
 }
 
 // Define a Collection that uses Animal as its document
-var Animals = new Mongo.Collection<AnimalDAO>("Animals", {
+var Animals = new Mongo.Collection<DBAnimal, AnimalDAO>("Animals", {
```

Comment:
This is the place where the naming may be crucial as this may be the only documented (so far) example of using this feature. As we have a limited version in one of our apps, I can suggest the naming we stick to right now:

```ts
// General pattern
const Xs = new Mongo.Collection<XDocument, X>(...);

// Example
const Animals = new Mongo.Collection<AnimalDocument, Animal>(...);`
```

While it's not a requirement and I understand where the _DAO_ suffix come from, I believe it should follow some convention. _(Maybe `AnimalPOD` will match `AnimalDAO`?)_